### PR TITLE
Add `plasma_physics` module for 0-D plasma physics tools

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Does this PR close or fix any Issues? Remember to create an Issue before starting work so that your fix / proposal can be addressed by the team.

e.g. Closes #{ID}

## Description

What is your PR trying to achieve? How did you go about achieving it?

## Interface Changes

If you've had to update an interface or introduce a new interface as part of your change then let us know here.

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Review by hsaunders1904 — APPROVED** [review]
Looks fine to me, just one typing correction

**Comment by CoronelBuendia** [comment]
> I had one thought about the location of this. If as the title suggests it just contains tools would a better location be a submodule `plasma_physics` to utilities? Or are we likely to run into dependency loops if we do that?

Tools is perhaps a bad word to have used... At the moment this might seem rather small to have as a module, but I anticipate growth - not least from #1086 

I don't know about dependency loops yet, and can't foresee what they would be like once this gets padded out with other stuff.

This seemed philosophically different enough from utilities, and specific enough to our "brief" to potentially warrant a new module. Unlike e.g. ML tools which are really generic and less directly relevant to our brief.

I'm not fussy about this though - right now it's barely used.

**Comment by je-cook** [comment]
Sure if its going to grow that makes sense, was just a thought

File: bluemira/plasma_physics/scaling_laws.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    Notes
+    -----
+    [2] ITER Physics Expert Group, Nucl. Fus. 39, 12, <https://iopscience.iop.org/article/10.1088/0029-5515/39/12/302/pdf>
+    equation (20)
+
+    \t:math:`\\tau_{E}=0.0562I_p^{0.93}B_t^{0.15}P_{sep}^{-0.69}n^{0.41}M^{0.19}R_0^{1.97}A^{-0.57}\\kappa^{0.78}`
+    """  # noqa :W505
+    I_p = I_p * 1e6  # [MA]
+    p_sep = p_sep * 1e6  # [MW]
+    n = n * 1e19  # [10^19/m^3]
```
Comment (je-cook) [inline bluemira/plasma_physics/scaling_laws.py:251]: same here actually, I think this is the wrong way round should probably do this:
```suggestion
    n = raw_uc(n, '1/m^3', '1e19/m^3')  # [10^19/m^3]
```

File: bluemira/plasma_physics/scaling_laws.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    \t:math:`B_{T}^{0.772 \\pm 0.031}a^{0.975 \\pm 0.08}R_{0}^{0.999 \\pm 0.101}`
+    """  # noqa :W505
+    law = PowerLawScaling(
+        constant=2.15e6,
+        constant_err=0.0,
+        exponents=[0, 0.782, 0.772, 0.975, 0.999],
+        exp_errs=[0.107, 0.037, 0.031, 0.08, 0.101],
+    )
+    leading_term = np.exp(1)
+    n_e20 = n_e / 1e20
```
Comment (je-cook) [inline bluemira/plasma_physics/scaling_laws.py:202]: possibly overkill but:
```suggestion
    n_e20 = raw_uc(n_e,  '1/m^3', '1e20/m^3')
```

**Review by je-cook — CHANGES_REQUESTED** [review]
Just some units things

**Review by je-cook — APPROVED** [review]
Great last test :smile:

**Comment by CoronelBuendia** [comment]
> Great last test smile

Actually come to think of the ITER confinement time should be around 3 seconds.. not 1! Probably messed up the inputs. I did it a bit whimsically...

**Pull request merged**